libs/@blockprotocol/graph/src/stdlib/bound.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  NonNullTimeInterval,
+  TemporalBound,
+} from "../types/temporal-versioning.js";
+
+export const compareBounds = (
+  lhs: TemporalBound,
+  rhs: TemporalBound,
+  lhsType: keyof NonNullTimeInterval,
+  rhsType: keyof NonNullTimeInterval,
```

Comment:
I think all these functions would be more immediately parseable if we didn't use `lhs`/`rhs` to name the arguments for all of them . 

Where order doesn't matter here, I think `firstBound`/`secondBound`, `leftHandSide`/`rightHandSide` also fine (I'd skip acronyms where possible as they introduce additional processing time for people who don't have lhs/rhs internalised). Or even `left` and `right`

We should call them something more descriptive where order does matter (I'll suggest elsewhere)

Can we add a docstring to this explaining that order doesn't matter and what it's for, i.e. sorting?

Reply:
I feel like `first` and `second` might actually imply _more_ of an order than "lhs" or "rhs" which are fairly common parameter choices for operators.

I generally completely avoid acronyms too, so I'm happy to rename them here but I think it really will inhibit readability of the function below. It adds a lot text which makes it harder to quickly see the actual logic that's happening here. (And the identity of the things does not matter, so really I _almost_ just want to call it `x` and `y`).

I have tried to highlight where parameter order _does_ matter, do you think it would help that much to add lines in all the documentation about ones where parameter _doesn't_ matter?

That said, I can add a doc-comment to this function, I opted not to just because it was a bog-standard JS comparison operation.

```ts
export const compareBounds = (
  leftHandSide: TemporalBound,
  rightHandSide: TemporalBound,
  leftHandSideType: keyof NonNullTimeInterval,
  rightHandSideType: keyof NonNullTimeInterval,
): number => {
  // If the bound values are not equal, then the bound with the lower value is less than the bound with the higher
  // value.
  if (leftHandSide.kind !== "unbounded" && rightHandSide.kind !== "unbounded") {
    if (leftHandSide.limit !== rightHandSide.limit) {
      return leftHandSide.limit.localeCompare(rightHandSide.limit);
    }
  }

  if (
    leftHandSide.kind === rightHandSide.kind &&
    leftHandSideType === rightHandSideType
  ) {
    return 0;
  }

  if (
    (leftHandSide.kind === "unbounded" && leftHandSideType === "start") ||
    (rightHandSide.kind === "unbounded" && rightHandSideType === "end") ||
    (leftHandSide.kind === "exclusive" &&
      rightHandSide.kind === "exclusive" &&
      leftHandSideType === "end" &&
      rightHandSideType === "start") ||
    (leftHandSide.kind === "exclusive" &&
      rightHandSide.kind === "inclusive" &&
      leftHandSideType === "end") ||
    (leftHandSide.kind === "inclusive" &&
      rightHandSide.kind === "exclusive" &&
      rightHandSideType === "start")
  ) {
    return -1;
  }

  if (
    (leftHandSide.kind === "unbounded" && leftHandSideType === "end") ||
    (rightHandSide.kind === "unbounded" && rightHandSideType === "start") ||
    (leftHandSide.kind === "exclusive" &&
      rightHandSide.kind === "exclusive" &&
      leftHandSideType === "start" &&
      rightHandSideType === "end") ||
    (leftHandSide.kind === "exclusive" &&
      rightHandSide.kind === "inclusive" &&
      leftHandSideType === "start") ||
    (leftHandSide.kind === "inclusive" &&
      rightHandSide.kind === "exclusive" &&
      rightHandSideType === "end")
  ) {
    return 1;
  }
  throw new Error(
    `Implementation error, failed to compare bounds.\nLHS: ${JSON.stringify(
      leftHandSide,
    )}\nRHS: ${JSON.stringify(rightHandSide)}`,
  );
};
```

Reply:
How about `left` and `right`? It's one more character. Or even `x` and `y`, sure.

Reply:
I don't despise `left` and `right` I guess 😅, does get rid of the acronym. I feel allergic to using `x` and `y` outside of code to do with co-ordinates or something tbh (I know I suggested it), so I've opted for `left` and `right`.